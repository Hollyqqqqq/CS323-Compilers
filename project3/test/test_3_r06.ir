FUNCTION hanoi :
PARAM v1
PARAM v2
PARAM v3
PARAM v4
IF v1 == #1 GOTO label1
GOTO label2
LABEL label1 :
t5 := v2 * #1000000
t4 := t5 + v4
WRITE t4
GOTO label3
LABEL label2 :
v5 := v1 - #1
ARG v3
ARG v4
ARG v2
ARG v5
t9 := CALL hanoi
t14 := v2 * #1000000
t13 := t14 + v4
WRITE t13
v9 := v1 - #1
ARG v4
ARG v2
ARG v3
ARG v9
t18 := CALL hanoi
LABEL label3 :
RETURN #0

FUNCTION main :
ARG #3
ARG #2
ARG #1
ARG #3
t23 := CALL hanoi
RETURN #0

