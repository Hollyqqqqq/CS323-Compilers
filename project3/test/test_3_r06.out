Program (1)
  IncludeList (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: hanoi
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: n
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: p1
            COMMA
            VarList (1)
              ParamDec (1)
                Specifier (1)
                  TYPE: int
                VarDec (1)
                  ID: p2
              COMMA
              VarList (1)
                ParamDec (1)
                  Specifier (1)
                    TYPE: int
                  VarDec (1)
                    ID: p3
        RP
      CompSt (1)
        LC
        DefList (1)
        StmtList (2)
          Stmt (2)
            IF
            LP
            Exp (2)
              Exp (2)
                ID: n
              EQ
              Exp (2)
                INT: 1
            RP
            Stmt (2)
              CompSt (2)
                LC
                DefList (1)
                StmtList (3)
                  Stmt (3)
                    Exp (3)
                      WRITE
                      LP
                      Exp (3)
                        Exp (3)
                          Exp (3)
                            ID: p1
                          MUL
                          Exp (3)
                            INT: 1000000
                        PLUS
                        Exp (3)
                          ID: p3
                      RP
                    SEMI
                  StmtList (1)
                RC
            ELSE
            Stmt (5)
              CompSt (5)
                LC
                DefList (1)
                StmtList (6)
                  Stmt (6)
                    Exp (6)
                      ID: hanoi
                      LP
                      Args (6)
                        Exp (6)
                          Exp (6)
                            ID: n
                          MINUS
                          Exp (6)
                            INT: 1
                        COMMA
                        Args (6)
                          Exp (6)
                            ID: p1
                          COMMA
                          Args (6)
                            Exp (6)
                              ID: p3
                            COMMA
                            Args (6)
                              Exp (6)
                                ID: p2
                      RP
                    SEMI
                  StmtList (7)
                    Stmt (7)
                      Exp (7)
                        WRITE
                        LP
                        Exp (7)
                          Exp (7)
                            Exp (7)
                              ID: p1
                            MUL
                            Exp (7)
                              INT: 1000000
                          PLUS
                          Exp (7)
                            ID: p3
                        RP
                      SEMI
                    StmtList (8)
                      Stmt (8)
                        Exp (8)
                          ID: hanoi
                          LP
                          Args (8)
                            Exp (8)
                              Exp (8)
                                ID: n
                              MINUS
                              Exp (8)
                                INT: 1
                            COMMA
                            Args (8)
                              Exp (8)
                                ID: p2
                              COMMA
                              Args (8)
                                Exp (8)
                                  ID: p1
                                COMMA
                                Args (8)
                                  Exp (8)
                                    ID: p3
                          RP
                        SEMI
                      StmtList (1)
                RC
          StmtList (10)
            Stmt (10)
              RETURN
              Exp (10)
                INT: 0
              SEMI
            StmtList (1)
        RC
    ExtDefList (13)
      ExtDef (13)
        Specifier (13)
          TYPE: int
        FunDec (13)
          ID: main
          LP
          RP
        CompSt (13)
          LC
          DefList (14)
            Def (14)
              Specifier (14)
                TYPE: int
              DecList (14)
                Dec (14)
                  VarDec (14)
                    ID: sum
                  ASSIGN
                  Exp (14)
                    INT: 3
              SEMI
            DefList (1)
          StmtList (15)
            Stmt (15)
              Exp (15)
                ID: hanoi
                LP
                Args (15)
                  Exp (15)
                    ID: sum
                  COMMA
                  Args (15)
                    Exp (15)
                      INT: 1
                    COMMA
                    Args (15)
                      Exp (15)
                        INT: 2
                      COMMA
                      Args (15)
                        Exp (15)
                          INT: 3
                RP
              SEMI
            StmtList (16)
              Stmt (16)
                RETURN
                Exp (16)
                  INT: 0
                SEMI
              StmtList (1)
          RC
      ExtDefList (1)
